src/Week1/HW01.hs

```diff
@@ -0,0 +1,53 @@
+{-# OPTIONS_GHC -Wall #-}
+module Week1.HW01 where
+
+-- Exercise 1 -----------------------------------------
+
+-- Get the last digit from a number
+lastDigit :: Integer -> Integer
+lastDigit =  flip mod 10
+
+-- Drop the last digit from a number
+dropLastDigit :: Integer -> Integer
+dropLastDigit x = (x - lastDigit x) `div` 10
+
+-- Exercise 2 -----------------------------------------
+
+toRevDigits :: Integer -> [Integer]
+toRevDigits x 
+    | x <= 0 = [] 
+    | otherwise = lastDigit x : toRevDigits(dropLastDigit x)
+
+-- Exercise 3 -----------------------------------------
+
+-- Double every second number in a list starting on the left.
+doubleEveryOther :: [Integer] -> [Integer]
+doubleEveryOther [] = []
+doubleEveryOther (x:xs)
+ | even (length xs) = 2 * x:doubleEveryOther xs
+ | otherwise = x:doubleEveryOther xs
+
+-- Exercise 4 -----------------------------------------
+
+-- Calculate the sum of all the digits in every Integer.
+sumDigits :: [Integer] -> Integer
+sumDigits xs = sum (foldr (++) [] (map toRevDigits xs))
```

Comment:
This can be done more simply (hint: find a function in [Data.List](https://hackage.haskell.org/package/base-4.21.0.0/docs/Data-List.html) that allows you to concatenate list of lists into a simple list)